### PR TITLE
Implement matrix division using CUSOLVER

### PR DESCRIPTION
**Pull request by xukai92**
Towards addressing https://github.com/JuliaGPU/CuArrays.jl/issues/440 via matrix division as the first step (https://github.com/JuliaGPU/CuArrays.jl/issues/440#issuecomment-537942479)

### CODE REVIEW

File: src/CuArrays.jl [hunk 1]
```diff
@@ -62,6 +62,7 @@ include("utils.jl")
 include("indexing.jl")
 include("broadcast.jl")
 include("matmul.jl")
+include("matinv.jl")
```
Comment (maleadt) [inline src/CuArrays.jl]: `matinv.jl` is pretty specific, maybe this functionality should go in `linalg.jl`, together with `matmul.jl`?

File: Manifest.toml [hunk 2]
```diff
@@ -74,9 +74,9 @@ version = "4.0.0"
 
 [[DataStructures]]
 deps = ["InteractiveUtils", "OrderedCollections"]
```
Comment (maleadt) [inline Manifest.toml]: if the manifest changes aren't needed, please leave them out

File: test/runtests.jl [hunk 3]
```diff
@@ -18,16 +18,16 @@ allowscalar(false)
 
 @testset "CuArrays" begin
 
-include("base.jl")
```
Comment (maleadt) [inline test/runtests.jl]: accidentally committed, I guess.

File: src/CuArrays.jl (hunk 1 above)
Comment (xukai92) [reply src/CuArrays.jl]: I thought `linalg.jl` is for overloading `LinearAlgebra` module. There is also a `matmul.jl` already existing. But I'm happy to move `matinv.jl` into `linalg.jl` - no problem.

File: Manifest.toml (hunk 2 above)
Comment (xukai92) [reply Manifest.toml]: I was not able to run tests without `] update` - no sure why. I can revert the change though.

Comment (maleadt) [reply Manifest.toml]: It is, but in Base `\(::Matrix, ::Matrix)` is implemented in the LinearAlgebra stdlib too, so I guess this makes sens?

File: src/CuArrays.jl (hunk 1 above)
Comment (maleadt) [reply src/CuArrays.jl]: It is, but in Base \(::Matrix, ::Matrix) is implemented in the LinearAlgebra stdlib too, so I guess this makes sens?

File: Manifest.toml (hunk 2 above)
Comment (xukai92) [reply Manifest.toml]: Oh I didn't release this! Let me move the code into `linalg.jl` then!

Comment (maleadt) [reply Manifest.toml]: You were using `julia --project` right? The manifest should always be valid, as it's what the CI infrastructure uses. Try getting rid of it, and we'll see what CI thinks about it :slightly_smiling_face:

File: src/CuArrays.jl (hunk 1 above)
Comment (xukai92) [reply src/CuArrays.jl]: Oh I didn't release this! Let me move the code into `linalg.jl` then!

File: Manifest.toml (hunk 2 above)
Comment (xukai92) [reply Manifest.toml]: I did with `--project=@.` inside the folder. I got errors about `CUstream` not defined.

Comment (maleadt) [reply Manifest.toml]: You probably didn't `instantiate` then (to have Pkg actually check out the versions in the Manifest).

Comment (xukai92) [reply Manifest.toml]: Oh I see. Thanks! I will revert this then.

**Comment by xukai92** [comment]
Changes done.

**Comment by maleadt** [comment]
bors try

**Pull request merged**